### PR TITLE
New branch to be able to make a PR to main

### PR DESCRIPTION
**Pull request by IsmaPi**
New PR because I had some issues with reopening a PR. This branch is a copy of a branch with a revamp of the whole repo. I changed the whole pipeline to ensure that all of it works, and it's what we talked about, creating a PR at the very beginning and then committing to it. The main changes are: full restructuring of the project layout, for example, models were moved from src/models/ to models/, created a utils/ and evaluation/ module, cleaned the repo a little, and deleted useless code (src/models_from_zero/, src/training/, scripts/visualize.py, old evaluation modules). Then i also rewrote both model architectures (torchmd_et.py, torchmd_gn.py) and I'm experimented with delta learning (analytical self-mobility skip connection + learned correction), zero-initialized output heads so training starts from the known physics baseline, and replaced the integer atom-type embedding with a continuous force projection. Furthermore, I implemented the PyTorch Lightning training wrapper (models/lightning_wrapper.py) with proper logging, validation metrics, learning rate scheduling, and early stopping. I also added automated evaluation (evaluation/pair_sweep.py, evaluation/single_step.py, evaluation/plot_training.py) that generates pair-interaction sweep plots, training curves, and single-step error metrics. The notebooks (visualization.ipynb, examples.ipynb) are leftovers from what we did and tested manually on Friday, used to verify that the data pipeline and model loading were working correctly. They are not part of the evaluation pipeline and can be removed. Regarding the pair sweep showing incorrect results: the models are currently not capturing cross-particle hydrodynamic interactions well, the self-mobility baseline (the dominant term) is learned correctly, but the interaction corrections are still being actively worked on since the results are promising, and I still need to work a bit more on it

### CODE REVIEW

File: evaluation/pair_sweep.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+    force_direction = force_direction / np.linalg.norm(force_direction)
+
+    # Create solver
+    if geometry == "nbody_open":
+        solver = lm.NBody(periodicityX="open", periodicityY="open", periodicityZ="open")
+        solver.initialize(viscosity=viscosity, hydrodynamicRadius=a)
+    elif geometry == "pse_periodic":
+        solver = lm.PSE(periodicityX="periodic", periodicityY="periodic", periodicityZ="periodic")
+        solver.setParameters(Lx=box_size, Ly=box_size, Lz=box_size, psi=1.0)
+        solver.initialize(viscosity=viscosity, hydrodynamicRadius=a)
```
Comment (RaulPPelaez) [inline evaluation/pair_sweep.py:67]: Adding this detail in here (the geometry) will bite you later. You will add more geometries to the arch and forget to modify this tiny function.

Instead, make this function receive two callables to compare, e.g. lambdas that take pos, force and return disp.

Then you pass the libmo solver and the model to this function.

Have a single source of truth for libmobility solver initialization. There should be only one place in your codebase in which initialize, setParameters, PSE and NBody are mentioned.

File: evaluation/pair_sweep.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+        )
+        data.batch = torch.zeros(2, dtype=torch.long, device=device)
+
+        pred_disp = model(data).cpu().numpy()
+        pred_all.append(pred_disp)
+
+    solver.clean()
+
+    import math
+    self_mobility = 1.0 / (6.0 * math.pi * viscosity * a)
```
Comment (RaulPPelaez) [inline evaluation/pair_sweep.py:106]: A simple way to remove this unit from the code altogether, is to choose viscosity = 1/(6pi), so that the self mobility is 1 always.

File: evaluation/plot_training.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
+"""Plot training curves from TensorBoard logs.
+
+Reads event files and generates matplotlib figures for loss, metrics,
+and learning rate over training.
+"""
```
Comment (RaulPPelaez) [inline evaluation/plot_training.py:5]: You take data from tensorboard and use matplotlib to plot it. That is a shame, given that tensorboard is already specialized in visualization. It literally gives you a website that updates live. See here https://www.tensorflow.org/tensorboard/get_started?hl=es-419

File: evaluation/single_step.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
+"""Single-step evaluation metrics.
+
+Computes MSE, MAE, and relative error between predicted and true displacements.
+No denormalization needed — model outputs physical units directly.
```
Comment (RaulPPelaez) [inline evaluation/single_step.py:4]: remove comment

File: models/lightning_wrapper.py [hunk 5]
```diff
@@ -0,0 +1,149 @@
+"""PyTorch Lightning module wrapping the hydrodynamic displacement models.
+
+Handles training/validation step, loss computation, optimizer/scheduler setup.
+Works with both HydroTorchMD_ET and HydroTorchMD_GN.
+"""
```
Comment (RaulPPelaez) [inline models/lightning_wrapper.py:5]: I am not seeing any mentions to a test loss around the project. Your trainers should always report a test loss at the end.

File: models/torchmd_gn.py
Comment (RaulPPelaez) [inline models/torchmd_gn.py:1]: GN is not going to get you very far. I would not even worry about it. TensorNet will probably give you much better results.

**Review by RaulPPelaez — COMMENTED** [review]
I gave it a review, the repo is in much better shape now, keep it up!
